lib/ansible/module_utils/urls.py

```diff
@@ -171,12 +171,98 @@
     except ImportError:
         HAS_MATCH_HOSTNAME = False
 
+HAS_CRYPTOGRAPHY = True
+try:
+    from cryptography import x509
+    from cryptography.hazmat.backends import default_backend
+    from cryptography.hazmat.primitives import hashes
+    from cryptography.exceptions import UnsupportedAlgorithm
+
+except ImportError:
+    HAS_CRYPTOGRAPHY = False
 
+GSSAPI_IMP_ERR = None
 try:
-    import urllib_gssapi
-    HAS_GSSAPI = True
```

Comment:
Good point, I've preserved the existing import and `HAS_GSSAPI` as it looks like the IPA modules in `community.general` rely on it to activate gssapi auth. Because `urllib_gssapi` depends on `gssapi`, `HAS_GSSAPI = True` means GSSAPI auth is available in `urls.py`. I've added a comment saying `urls.py` does not use `HAS_GSSAPI` but it's kept there for backwards compatibility.